janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/NumericTranslationQueryParser.java

```diff
@@ -0,0 +1,175 @@
+                return getNumericRangeQuery(field, dataType, start, end, includeLower, includeUpper);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + field + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.getRangeQuery(field, start, end, includeLower, includeUpper);
+    }
+
+    @Override
+    protected Query newTermQuery(Term t) {
+        if (t.field() == null) {
+            return super.newTermQuery(t);
+        }
+
+        Class<?> dataType = storeRetriever.get(t.field()).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return buildNumericQuery(t.field(), t.text(), dataType);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + t.field() + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.newTermQuery(t);
+    }
+
+    @Override
+    protected Query newWildcardQuery(final Term t) {
+        if (t.field() == null) {
+            return super.newTermQuery(t);
+        }
+
+        Class<?> dataType = storeRetriever.get(t.field()).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return buildNumericQuery(t.field(), t.text(), dataType);
+            } catch (NumberFormatException e) {
+                log.warn("Expected Number type for " + t.field() + ":" + dataType + ", but " + e.getMessage());
+            }
+        }
+
+        return super.newWildcardQuery(t);
+    }
+
+    private Query getNumericRangeQuery(final String field, final Class<?> type, String start, String end, final boolean includeLower,
+                                       final boolean includeUpper) {
+        if (type.equals(Integer.class)) {
+            Integer min = isMatchAll(start) ? Integer.MIN_VALUE : Integer.parseInt(start);
+            Integer max = isMatchAll(end) ? Integer.MAX_VALUE : Integer.parseInt(end);
+            if (!includeLower) {
+                min = Math.addExact(min, 1);
+            }
+            if (!includeUpper) {
+                max = Math.addExact(max, -1);
+            }
+            return IntPoint.newRangeQuery(field, min, max);
+        }
+
+        if (type.equals(Long.class)) {
+            Long min = isMatchAll(start) ? Long.MIN_VALUE : Long.parseLong(start);
+            Long max = isMatchAll(end) ? Long.MAX_VALUE : Long.parseLong(end);
+            if (!includeLower) {
+                min = Math.addExact(min, 1);
+            }
+            if (!includeUpper) {
+                max = Math.addExact(max, -1);
+            }
+            return LongPoint.newRangeQuery(field, min, max);
+        }
+
+        if (type.equals(Float.class)) {
+            Float min = isMatchAll(start) ? Float.NEGATIVE_INFINITY : Float.parseFloat(start);
+            Float max = isMatchAll(end) ? Float.MAX_VALUE : Float.parseFloat(end);
```

Comment:
I didn't check the logic but why do you use `NEGATIVE_INFINITY` for `min` and `MAX_VALUE` for `max`?
Should we use `POSITIVE_INFINITY` for `max` or `MIN_VALUE` for `min`?

Reply:
This is typo